routemaster/config/model.py

```diff
@@ -45,7 +45,13 @@ def applies(path: Sequence[str], d: Dict[str, Any]) -> bool:
         return applies(self.metadata_path.split('.'), update)
 
 
-Trigger = Union[TimeTrigger, IntervalTrigger, MetadataTrigger]
+class OnEntryTrigger:
+    """Trigger on entry to a given gate."""
+    def __eq__(self, other):
+        return isinstance(other, OnEntryTrigger)
```

Comment:
Related: should we also be overriding `__hash__`?

Reply:
451d8d3fc227650860489644ddfc061c3634b1eb